File: app/src/main/java/com/example/sercanyusuf/boomsetdemo/di/components/AppComponent.java
package com.example.sercanyusuf.boomsetdemo.di.components;


import com.example.sercanyusuf.boomsetdemo.MainActivity;
import com.example.sercanyusuf.boomsetdemo.di.modules.AppModule;

import javax.inject.Singleton;

import dagger.Component;


@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {

    void inject(MainActivity boomsetApp);
    // void inject (MainActivity target);

}

File: app/src/main/java/com/example/sercanyusuf/boomsetdemo/di/components/AppComponent.kt
package com.example.sercanyusuf.boomsetdemo.di.components


import android.app.Application
import com.example.sercanyusuf.boomsetdemo.di.modules.AppModule
import com.example.sercanyusuf.boomsetdemo.di.modules.LocalModule
import com.example.sercanyusuf.boomsetdemo.di.modules.NetModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class, LocalModule::class,NetModule::class))
interface AppComponent {
    fun inject(BoomsetApp: Application)
}